src/server_manager/web_app/karma.conf.js

```diff
@@ -18,15 +18,22 @@ const baseConfig = makeConfig({
   defaultMode: 'development'
 });
 
+const test_patterns = [
+  '**/*.spec.ts',
+  // We need to test data_formatting in a browser context
+  '../infrastructure/data_formatting.spec.ts'
```

Comment:
Why do  we need to test the formatting in a browser context?  It doesn't use the DOM.